recognition/scoring.py

```diff
@@ -235,29 +250,33 @@ def tasks(self):
         yield Task("run", mini_task=True)
 
     def run(self, move_files=True):
-        hubscr_path = (
-            os.path.join(gs.SCTK_PATH, "hubscr.pl")
-            if hasattr(gs, "SCTK_PATH")
-            else "hubscr.pl"
-        )
-        sctk_opt = ["-p", gs.SCTK_PATH] if hasattr(gs, "SCTK_PATH") else []
+        sctk_path = ""
+        if self.sctk_binary_path is not None:
+            sctk_path = self.sctk_binary_path.get_path()
+        elif hasattr(gs, "SCTK_PATH"):
+            sctk_path = gs.SCTK_PATH
+        hubscr_path = os.path.join(
+            sctk_path, "hubscr.pl"
+        )  # evaluates to just "hubscr.pl" if sctk_path is empty
+
+        sctk_opt = ["-p", sctk_path] if sctk_path else []
 
         ref = self.ref
         try:
-            ref = shutil.copy(tk.uncached_path(ref), ".")
+            ref = shutil.copy(ref.get_path(), ".")
         except shutil.SameFileError:
```

Comment:
I am wondering: why are we copying ref and hyp?

Reply:
how can this be the same file?? `"."` should be the job work dir, correct? ergo this should not be the same file

Reply:
> why are we copying ref and hyp?

dunno.

> how can this be the same file??

maybe during automatic LM scale optimization?